Source/Core/DolphinQt/RenderWidget.cpp

```diff
@@ -254,6 +254,10 @@ void RenderWidget::SetCursorLocked(bool locked, bool follow_aspect_ratio)
   if (locked)
   {
 #ifdef _WIN32
+    // This will prevent the mouse from interacting with the task bar
+    // while in windowed / borderless fullscreen.
+    SetWindowPos((HWND) winId(), HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
```

Comment:
Have you checked if no other code changes the HWND_TOPMOST on the render widget/window handle? We don't want any conflicts there.

Reply:
And could you explain the exact benefit of NOACTIVATE?

Reply:
> Have you checked if no other code changes the HWND_TOPMOST on the render widget/window handle? We don't want any conflicts there.

My additions of `SetWindowPos` are the only two in the project file (aka DolphinQt folder).

> And could you explain the exact benefit of NOACTIVATE?

per my original post, 
> the last flag is important to prevent a loop with popup boxes fighting for focus.

The default behaviour of this function is to omit an `activate` event to the window, which will focus the window if not already. This creates a deadlock with any popup dialogs, such as the "Are you sure you want to exit the emulation" pop up, which also wants focus. They will continue to fight over who has focus until Dolphin is forcefully closed. There isn't any reason to trigger an activate event, this function is just used to set the window Z position, so removing it would be ideal anyway.

Reply:
I still don't really understand what you mean, as there was no problem like that with the exit confirmation window in my tests, but anyway, this sounds like a general improvement, actually unrelated from mouse locking (which is an optional feature)? So should the SWP_NOACTIVATE only be here?

Reply:
> 
> 
> I still don't really understand what you mean, as there was no problem like that with the exit confirmation window in my tests, but anyway, this sounds like a general improvement, actually unrelated from mouse locking (which is an optional feature)? So should the SWP_NOACTIVATE only be here?

This problem definitely exists, the only way I could close Dolphin was by literally *logging off*, because the focus stealing was so intense I couldn't even end the process in task manager, and the problem would happen every time until I added the `SWP_NOACTIVATE` flag. Regardless of if you were able to successfully recreate it, the problem can happen and therefore this change should exist to prevent it, especially since there are no downsides. 

The point is, there is no reason to activate the window here, regardless of if the problem does or doesn't exist. This is only to change the window's level, we do not need to simulate the same events that occur when a user adjusts the window.

Reply:
True, the SWP_NOACTIVATE here is fine, but are you saying that the "focus stealing" but between the render and exit widgets already existed before this? Could you provide a video for that?

If you are interested, I think there is also a but that happens when going full screen, like about 1 time out of 3, full-screen doesn't get the keyboard focus and pressing exit does not open the confirmation widget, or maybe it does but it doesn't put it in the foreground, or something like that.